tinygrad/codegen/kernel.py

```diff
@@ -27,6 +27,7 @@ def __str__(self): return f"tensor_core<{self.device}, {self.dims}, {self.dtype_
   ],
   "HIP": [
     TensorCore(device="HIP", dims=[16,16,16], dtype_in=dtypes.half, dtype_out=dtypes.float, threads=[16,2], thread_local_sizes=[16,16,8], thread_local_aliases=[ [[-1], [0], [1]], [[-1], [1], [0]], [[0], [1], [2, -1]] ]),
+    TensorCore(device="HIP", dims=[16,16,16], dtype_in=dtypes.half, dtype_out=dtypes.half,  threads=[16,2], thread_local_sizes=[16,16,8], thread_local_aliases=[ [[-1], [0], [1]], [[-1], [1], [0]], [[0], [1], [2, -1]] ]),
```

Comment:
You wouldn't be able to use `__builtin_amdgcn_wmma_f32_16x16x16_f16_w32` for this, but I don't see gating in cstyle

Reply:
It actually works fine (auto-casts properly - same goes for FP16 in METAL), but yes I have the WMMA ops passing the dtype_in/dtype_out to be able to gate in cstyle when necessary.